core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -66,8 +78,45 @@
     public final static String INDEX = ".ingest";
     public final static String TYPE = "pipeline";
 
+    final static Settings INGEST_INDEX_SETTING = Settings.builder()
+        .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+        .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 1)
+        .put("index.mapper.dynamic", false)
+        .build();
+
+    final static String PIPELINE_MAPPING;
+
+    static {
+        try {
+            PIPELINE_MAPPING = XContentFactory.jsonBuilder().startObject()
+                .field("dynamic", "strict")
+                .startObject("_all")
+                    .field("enabled", false)
+                .endObject()
+                .startObject("properties")
+                    .startObject("processors")
```

Comment:
We don't have control over the amount pipelines and size of each pipeline that will be stored. When there are a couple of pipelines I think it is fine to store it in the cluster state (would make things easier to manage for ingest as well). But when there are many pipelines then I don't feel that the cluster state is the right place to store the configuration.